server/src/test/java/org/elasticsearch/action/search/ParsedScrollIdTests.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.search.internal.ShardSearchContextId;
+import org.elasticsearch.test.ESTestCase;
+
+public class ParsedScrollIdTests extends ESTestCase {
+    public void testHasLocalIndices() {
```

Comment:
This test relies on the `TransportSearchHelperTests#testParseScrollId` , where, given `SearchShardTargets`, the scroll id is encoded and then decoded, and the test asserts that the "cluster alias" shows up after decoding.

IMO, this test adds little value on top of that, and it's coded away from the above mentioned piece of code.
My personal approach would've been to parametrise the `TransportSearchHelperTests#generateQueryResults` method to allow it to generate scroll ids with/without cluster aliases, and assert the `ParsedScrollId#hasLocalIndices` method there as well (probably in another test but not necessarily so).

Reply:
I am not sure if I follow your comment. This test is an unit test for `ParsedScrollId` and has nothing to do with `TransportSearchHelperTests#testParseScrollId `. Maybe you were commenting on a previous version of this test, i.e. the one before I moved this method as part of `ParsedScrollId` following your comment of [OO design](https://github.com/elastic/elasticsearch/pull/62053#discussion_r484745362)?